src/history/hs.c

```diff
@@ -1214,10 +1217,17 @@ __hs_delete_key_from_pos(
         upd->start_ts = upd->durable_ts = WT_TS_NONE;
         WT_ERR(__wt_hs_modify(hs_cbt, upd));
         upd = NULL;
+        WT_STAT_CONN_INCR(session, cache_hs_remove_key_truncate);
     }
     if (ret == WT_NOTFOUND)
-        return (0);
+        ret = 0;
 err:
+    if (removed) {
+        if (mixed_ts)
+            WT_STAT_CONN_INCR(session, cache_hs_key_truncate_mix_ts);
```

Comment:
Can we update these in the callers instead of passing mixed_ts

Reply:
If we do that, we can't tell whether we ended up deleting a record.

But after thinking about it, this is probably better. In some code paths, we have to do `search_near` to tell whether we need to delete anything. If we're doing this a lot, we'd probably like to know about it in the statistics.